### PR TITLE
Update conformance report for NGINX Kubernetes Gateway

### PR DESCRIPTION
**Pull request by sjberman**
**What type of PR is this?**

/area conformance

**What this PR does / why we need it**:

Updates conformance profile for NGINX Kubernetes Gateway.

**Which issue(s) this PR fixes**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: conformance/reports/v0.8.0/nginxinc-nginx-kubernetes-gateway.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+apiVersion: gateway.networking.k8s.io/v1alpha1
+date: "2023-08-31T16:40:14Z"
+gatewayAPIVersion: TODO
```
Comment (robscott) [inline conformance/reports/v0.8.0/nginxinc-nginx-kubernetes-gateway.yaml:3]: Unfortunately we still need to manually set this
```suggestion
gatewayAPIVersion: v0.8.0
```

**Review by robscott — COMMENTED** [review]
Thanks @sjberman!

/ok-to-test
/approve

**Comment by robscott** [comment]
Thanks @sjberman! One last thing I forgot to mention, if you want to update your implementations entry to point to this later conformance report, you can copy what Contour did: https://github.com/kubernetes-sigs/gateway-api/pull/2364/files#diff-e7134a1e035e078a0323e26594afa0955026442dbf79173af892dd449aac56a2R157.

That's non-blocking and can be in a follow up though, up to you. Feel free to remove hold if you want to cover that in a follow up PR.

/lgtm
/hold

**Comment by sjberman** [comment]
@robscott Thanks for the suggestion. Done.

**Comment by robscott** [comment]
Thanks!

/lgtm
/hold cancel

**Pull request merged**